include/seqan3/io/sequence/sequence_file_in.hpp

```diff
@@ -1,175 +1,911 @@
+                                    typename traits_type::id_alphabet>;
+    //!\brief The type of field::QUAL (std::vector <seqan3::phred42> by default).
+    using quality_type          = typename traits_type::template quality_container<
+                                    typename traits_type::quality_alphabet>;
+    //!\brief The type of field::SEQ_QUAL (std::vector <seqan3::dna5q> by default).
+    using sequence_quality_type = typename traits_type::
+                                    template sequence_container<quality_composition<typename traits_type::sequence_alphabet,
+                                                                                    typename traits_type::quality_alphabet>>;
 
-    // constructor with arg
-    sequence_file_in(std::string const & _file_name);
+    //!\brief The previously defined types aggregated in a seqan3::type_list.
+    using field_types       = type_list<sequence_type, id_type, quality_type, sequence_quality_type>;
 
-    // copy construction and assignment are deleted
-    // implicitly because we don't want multiple access to file
+    //!\brief The type of the record, a specialisation of seqan3::record; acts as a tuple of the selected field types.
+    using record_type       = record<detail::select_types_with_ids_t<field_types, field_ids, selected_field_ids>,
+                                     selected_field_ids>;
+    //!\}
+
+    /*!\name Field column types
+     * \brief These types may be manipulated via the \ref traits_type to achieve different storage behaviour.
+     * \{
+     */
+    //!\brief Column type of field::SEQ; usually seqan3::concatenated_sequences<sequence_type>.
+    using sequence_column_type          = typename traits_type::template sequence_container_container<sequence_type>;
+    //!\brief Column type of field::ID; usually seqan3::concatenated_sequences<id_type>.
+    using id_column_type                = typename traits_type::template id_container_container<id_type>;
+    //!\brief Column type of field::QUAL; usually seqan3::concatenated_sequences<quality_type>.
+    using quality_column_type           = typename traits_type::template quality_container_container<quality_type>;
+    //!\brief Column type of field::SEQ_QUAL; usually seqan3::concatenated_sequences<sequence_quality_type>.
+    using sequence_quality_column_type  = typename traits_type::template id_container_container<sequence_quality_type>;
+    //!\brief The previously defined types aggregated in a seqan3::type_list.
+    using column_types                  = type_list<sequence_column_type,
+                                                    id_column_type,
+                                                    quality_column_type,
+                                                    sequence_quality_column_type>;
+    //!\brief The type of the internal columns buffer.
+    using columns_type = record<detail::select_types_with_ids_t<column_types, field_ids, selected_field_ids>,
```

Comment:
All of these terms are just more confusing, because "range" is used in all sorts of contexts. "column" is much more specific. Like I said, I will add general file documentation that iluistrates this more clearly.

Reply:
I agree, that range might not be so good after all.
Well, not completely sure ... just names ...
On the other hand, the terminology is either row/column or record/field, right? Just that it is hard to express, that all elements from a field are read. 😕 
Guess, we keep it like this.